cc/attestation/cose.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2024 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CC_ATTESTATION_COSE_H_
+#define CC_ATTESTATION_COSE_H_
+
+#include <memory>
+#include <string>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "libcppbor/include/cppbor/cppbor.h"
+#include "libcppbor/include/cppbor/cppbor_parse.h"
+
+namespace oak::attestation {
+
+// COSE Sign1 object.
+// <https://datatracker.ietf.org/doc/html/rfc8152#section-4.2>
+class CoseSign1 {
+ public:
+  const cppbor::Bstr* protected_headers;
+  const cppbor::Map* unprotected_headers;
+  const cppbor::Bstr* payload;
+  const cppbor::Bstr* signature;
+
+  CoseSign1(const cppbor::Bstr* protected_headers, const cppbor::Map* unprotected_headers,
+            const cppbor::Bstr* payload, const cppbor::Bstr* signature,
+            std::unique_ptr<cppbor::Item>&& item)
+      : protected_headers(protected_headers),
+        unprotected_headers(unprotected_headers),
+        payload(payload),
+        signature(signature),
+        item_(std::move(item)) {}
+
+  static absl::StatusOr<CoseSign1> Deserialize(const std::vector<uint8_t>& data);
+
+ private:
+  // Parsed CBOR item containing COSE Sign1 object.
+  std::unique_ptr<cppbor::Item> item_;
+};
+
+// COSE Key object.
+// <https://www.rfc-editor.org/rfc/rfc8152#section-7>
+class CoseKey {
+ public:
+  const cppbor::Uint* kty;
+  const cppbor::Bstr* kid;
+  const cppbor::Nint* alg;
+  const cppbor::Array* key_ops;
+
+  const cppbor::Uint* crv;
+  const cppbor::Bstr* x;
+
+  CoseKey(const cppbor::Uint* kty, const cppbor::Bstr* kid, const cppbor::Nint* alg,
+          const cppbor::Array* key_ops, const cppbor::Uint* crv, const cppbor::Bstr* x,
+          std::unique_ptr<cppbor::Item>&& item)
```

Comment:
Some documentation is needed - what do all these parameters mean?

Reply:
Updated the docs